JavaComparator/Comparator.java

```diff
@@ -0,0 +1,133 @@
+import java.util.ArrayList;
```

Comment:
Please send the changes for review after incorporating the following comments in the file in all java files.